src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs

```diff
@@ -159,6 +159,12 @@ IFormManager formManager
             return RedirectToAction("Edit", new { workflowDefinitionRecord.Id });
         }
 
+        public JsonResult State(int? id) {

```

Comment:
If it's a GET then we need to check for the permissions:
`if (!Services.Authorizer.Authorize(StandardPermissions.SiteOwner, T("Not authorized to edit workflows")))
                 return new HttpUnauthorizedResult();`